bindings/node/index.d.ts

```diff
@@ -1,8 +1,19 @@
 import type { Document, Binary } from 'bson';
-import type { MongoClient, BulkWriteResult, ClientSession } from 'mongodb';
+import type { MongoClient, BulkWriteResult, ClientSession, DeleteResult, FindCursor } from 'mongodb';
 
 export type ClientEncryptionDataKeyProvider = 'aws' | 'azure' | 'gcp' | 'local' | 'kmip';
 
+export interface DataKey {
+  _id: Binary;
```

Comment:
Just a suggestion and feel free to ignore, but for this one it could make sense to go one step further:

```suggestion
  _id: Binary & { sub_type: typeof Binary.SUBTYPE_UUID };
```

Reply:
<img width="828" alt="Screen Shot 2022-07-15 at 10 27 18 AM" src="https://user-images.githubusercontent.com/23407842/179244292-e2471fc5-f766-46b0-bf33-3124bd437081.png">

This seems to cause errors, I don't think our Binary type is typed well enough for this.  The `sub_type` is a runtime value on Binary and we don't have a way to relate the `number` in the Binary constructor to the proper subtype.

Separate issue - I noticed that I don't see any type errors in the `index.d.ts` in libmongocrypt.  I added `const foo: string = 3;` to the index and vscode didn't complain.  All the more reason to convert our bindings to Typescript I guess!